src/Symfony/Component/Notifier/Bridge/GoogleChat/GoogleChatOptions.php

```diff
@@ -82,18 +81,41 @@ public function text(string $text): static
     }
 
     /**
+     * @param string $threadKey Opaque thread identifier string that can be specified to group messages into a single thread.
+     *                          If this is the first message with a given thread identifier, a new thread is created.
+     *                          Subsequent messages with the same thread identifier will be posted into the same thread.
+     *                          {@see https://developers.google.com/workspace/chat/api/reference/rest/v1/spaces.messages#thread}
+     *
      * @return $this
      */
-    public function setThreadKey(?string $threadKey): static
+    public function threadKey(string $threadKey): static
     {
-        $this->threadKey = $threadKey;
+        $this->options['thread']['threadKey'] = $threadKey;
 
         return $this;
     }
 
+    /**
+     * @return $this
+     *
+     * @deprecated since Symfony 7.2, to be removed in 8.0
```

Comment:
Why do you need to deprecate this method? The new function `threadKey` serves the same purpose.

Reply:
This change was made to adopt the same "naming convention" of the other methods in this class.

The new `threadKey()` method served the same purpose because it was a simple rename of the setter and I wanted to keep the backward compatibility here.

This has been reverted back and the setter has been kept.